python_modules/dagster/dagster_tests/core_tests/instance_tests/test_instance_reexecution.py

```diff
@@ -0,0 +1,100 @@
+import os
+
+import pytest
+
+from dagster import DagsterInstance, execute_pipeline, job, op, reconstructable, repository
+from dagster.core.storage.pipeline_run import PipelineRunStatus
+from dagster.core.storage.tags import RESUME_RETRY_TAG
+from dagster.core.test_utils import (
+    instance_for_test,
+    poll_for_finished_run,
+    step_did_not_run,
+    step_succeeded,
+)
+from dagster.core.workspace.context import WorkspaceProcessContext
+from dagster.core.workspace.load_target import PythonFileTarget
+from dagster.seven import get_system_temp_directory
+
+CONDITIONAL_FAIL_FILE = os.path.join(
+    get_system_temp_directory(), "chained_failure_pipeline_conditionally_fail"
+)
+
+
+@op
+def before_failure():
+    return "hello"
+
+
+@op
+def conditional_fail(_, input_value):
+    if os.path.isfile(os.path.join(get_system_temp_directory(), CONDITIONAL_FAIL_FILE)):
+        raise Exception("blah")
+
+    return input_value
+
+
+@op
+def after_failure(_, input_value):
+    return input_value
+
+
+@job
+def conditional_fail_job():
+    after_failure(conditional_fail(before_failure()))
+
+
+@repository
+def repo():
+    return [conditional_fail_job]
+
+
+@pytest.fixture
+def instance() -> DagsterInstance:
+    with instance_for_test() as instance:
+        yield instance
+
+
+@pytest.fixture
+def workspace(instance):
+    with WorkspaceProcessContext(
+        instance,
+        PythonFileTarget(
+            python_file=__file__,
+            attribute=None,
+            working_directory=None,
+            location_name="repo_loc",
+        ),
+    ) as workspace_process_context:
+        yield workspace_process_context.create_request_context()
+
+
+def test_create_reexecuted_run_from_failure(instance: DagsterInstance, workspace):
+    try:
+        # trigger failure in the conditionally_fail solid
+        with open(CONDITIONAL_FAIL_FILE, "w", encoding="utf8"):
+            result = execute_pipeline(reconstructable(conditional_fail_job), instance=instance)
+    finally:
+        os.remove(CONDITIONAL_FAIL_FILE)
```

Comment:
could use env var via `environ` context manager instead of FS. I just worry about test flakiness with FS stuff.